Realm/NSError+RLMSync.h

```diff
@@ -32,6 +32,13 @@ NS_ASSUME_NONNULL_BEGIN
 - (nullable void(^)(void))rlmSync_clientResetBlock NS_REFINED_FOR_SWIFT;
 
 /**
+ Given a Realm Object Server permission denied error, return the block that
+ can be called to manually initiate or cancel the Realm file deletion process,
+ or nil if the error isn't a permission denied error.
+ */
```

Comment:
Attaching functionality to a NSError like this is really weird and unusual, but I guess since client reset is already done that way this should as well (at least for now).